### PR TITLE
make the search queries #11

### PR DESCRIPTION
**Pull request by amanshawar**

### CODE REVIEW

File: src/handler.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 const path = require('path');
 const fs = require('fs');
-//const getData=require('../queries/getData.js');
+var request1 = require('request');
```
Comment (m4v15) [inline src/handler.js]: change to `const`

File: src/handler.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
+var url = require('url');
```
Comment (m4v15) [inline src/handler.js]: `const`

File: src/handler.js (hunk 1 above)
Comment (m4v15) [reply src/handler.js]: also `requestModule` is a better name

File: src/handler.js [hunk 3]
```diff
@@ -45,6 +49,21 @@ const  publicHandler=(request,response)=>{
 //------------------------------------------------
 const searchHandler=(request,response)=>{
 
+  var { query }=url.parse(request.url);
+  var {category}=queryString.parse(query);
+  console.log(category);
```
Comment (m4v15) [inline src/handler.js]: remove `console.log`

File: src/handler.js [hunk 4]
```diff
@@ -45,6 +49,21 @@ const  publicHandler=(request,response)=>{
...
+  var {text}=queryString.parse(query);
+console.log(text);
```
Comment (m4v15) [inline src/handler.js]: remove

File: src/server.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 const http = require('http');
 const router = require('./router.js');
 const server = http.createServer(router);
-const PORT = process.env.PORT || 3001;
+const PORT = process.env.PORT || 3003;
```
Comment (m4v15) [inline src/server.js:4]: leish?

**Comment by amanshawar** [comment]
change port!
change the data in tables in db
update searchhandler
make getdata.js

File: queries/getData.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+const dbConnection = require('../database/db_connection.js');
+
+  const getData = (category,text,cb) => {
+    if (category==="song")
+    {
+      var q=`SELECT name,img_url,title,link from artists join track_artist on artists.id=track_artist.artist_id join tracks on tracks.id=track_artist.track_id where title='${text}'`;
```
Comment (m4v15) [inline queries/getData.js]: You should be using parameterised queries: https://node-postgres.com/features/queries#parameterized-query

**Review by Samaamro20 — APPROVED** [review]
well Done :+1:

File: queries/getData.js (hunk 6 above)
Comment (m4v15) [reply queries/getData.js]: so this should be 
```js
      const SQLString=`SELECT name,img_url,title,link from artists join track_artist on artists.id=track_artist.artist_id join tracks on tracks.id=track_artist.track_id where title=$1`;
      const parameters = [text]
```

And then the call to the database should be:

```js
dbConnection.query(SQLString, parameters, (err, res) => {
        if (err) return cb(err);
        cb(null, res.rows);
      });
```

File: queries/getData.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      dbConnection.query(q, (err, res) => {
+        if (err) return cb(err);
+        cb(null, res.rows);
+      });
+    }
+    else if (category==="artist") {
+      var q=`select artists.name,artists.img_url,tracks.title,tracks.link from artists full outer join track_artist on track_artist.artist_id=artists.id full outer JOIN tracks on track_artist.track_id=tracks.id where artists.name ='${text}';`
```
Comment (m4v15) [inline queries/getData.js]: same as above

**Review by m4v15 — CHANGES_REQUESTED** [review]
you need to add `request` to your `package.json`

File: src/server.js (hunk 5 above)
Comment (m4v15) [reply src/server.js:4]: https://stackoverflow.com/a/9346231/9256400

Comment (m4v15) [reply src/server.js:4]: I'm assuming the port `3001` was in use on your computer so it wouldn't run. When this happens you **should not** just change the port on master - what if on @Hadeel-Salamin or @Samaamro20 's laptop's, the port `3003` is in use. you would have to keep changing it in every PR.

Instead, use the link above to learn how to kill a process in a port (maybe you have your server running in another window, or sometimes ports just need killing 💀 )

**Review by m4v15 — APPROVED** [review]
Notes for future pull requests:

- Make sure you are using `const` not var
- use better variable names that describe the variable
- delete `console.logs`

**Review by Hadeel-Salamin — APPROVED** [review]

**Pull request merged**